web/src/components/storage/ProposalPage.tsx

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) [2022-2024] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useReducer, useEffect, useRef } from "react";
+import { Grid, GridItem, Stack } from "@patternfly/react-core";
+import { Page, Drawer } from "~/components/core/";
+import ProposalTransactionalInfo from "./ProposalTransactionalInfo";
+import ProposalSettingsSection from "./ProposalSettingsSection";
+import ProposalResultSection from "./ProposalResultSection";
+import ProposalActionsSummary from "~/components/storage/ProposalActionsSummary";
+import { ProposalActionsDialog } from "~/components/storage";
+import { _ } from "~/i18n";
+import { SPACE_POLICIES } from "~/components/storage/utils";
+import { toValidationError, useCancellablePromise } from "~/utils";
+import { useIssues } from "~/queries/issues";
+import { IssueSeverity } from "~/types/issues";
+import {
+  useAvailableDevices,
+  useDeprecated,
+  useDeprecatedChanges,
+  useDevices,
+  useProductParams,
+  useProposalMutation,
+  useProposalResult,
+  useVolumeDevices,
+  useVolumeTemplates,
+} from "~/queries/storage";
+import { probe } from "~/api/storage";
+
+/**
+ * Which UI item is being changed by user
+ */
+export const CHANGING = Object.freeze({
+  ENCRYPTION: Symbol("encryption"),
+  TARGET: Symbol("target"),
+  VOLUMES: Symbol("volumes"),
+  POLICY: Symbol("policy"),
+  BOOT: Symbol("boot"),
+});
+
+// mapping of not affected values for settings components
+// key:   component name
+// value: list of items which can be changed without affecting
+//        the state of the component
+export const NOT_AFFECTED = {
+  // the EncryptionField shows the skeleton only during initial load,
+  // it does not depend on any changed item and does not show skeleton later.
+  // the ProposalResultSection is refreshed always
+  InstallationDeviceField: [CHANGING.ENCRYPTION, CHANGING.BOOT, CHANGING.POLICY, CHANGING.VOLUMES],
+  PartitionsField: [CHANGING.ENCRYPTION, CHANGING.POLICY],
+  ProposalActionsSummary: [CHANGING.ENCRYPTION, CHANGING.TARGET],
+};
```

Comment:
That reminds me I'd like to look for another approach for this now that the UI has changed A LOT.

Reply:
Yes, I agree. I wanted to get rid of all this stuff but there are already too many changes 😅